src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,943 @@
+        strip = strip.crop(scroll_x, scroll_x + self.scrollable_content_region.width)
+
+        highlighted = self.highlighted
+        mouse_over = self._mouse_hovering_over
+        spans = self._spans
+
+        # Handle drawing a disabled option.
+        if self._options[option_index].disabled:
+            # Disabled but the highlight?
+            if option_index == highlighted:
+                return strip.apply_style(
+                    self.get_component_rich_style(
+                        "option-list--option-hover-highlighted-disabled"
+                        if option_index == mouse_over
+                        else "option-list--option-highlighted-disabled"
+                    )
+                )
+            # Disabled but mouse hover?
+            if option_index == mouse_over:
+                return strip.apply_style(
+                    self.get_component_rich_style("option-list--option-hover-disabled")
+                )
+            # Just a normal disabled option.
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-disabled")
+            )
+
+        # Handle drawing a highlighted option.
+        if highlighted is not None and line_number in spans[highlighted]:
+            # Highlighted with the mouse over it?
+            if option_index == mouse_over:
+                return strip.apply_style(
+                    self.get_component_rich_style(
+                        "option-list--option-hover-highlighted"
+                    )
+                )
+            # Just a normal highlight.
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-highlighted")
+            )
+
+        # Perhaps the line is within an otherwise-uninteresting option that
+        # has the mouse hovering over it?
+        if mouse_over is not None and line_number in spans[mouse_over]:
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-hover")
+            )
+
+        # It's a normal option line.
+        return strip.apply_style(self.rich_style)
+
+    def scroll_to_highlight(self) -> None:
+        """Ensure that the highlighted option is in view."""
+        highlighted = self.highlighted
+        if highlighted is None:
+            return
+        try:
+            span = self._spans[highlighted]
+        except IndexError:
+            # Index error means we're being asked to scroll to a highlight
+            # before all the tracking information has been worked out.
+            # That's fine; let's just NoP that.
+            return
+        self.scroll_to_region(
+            Region(
+                0, span.first, self.scrollable_content_region.width, span.line_count
+            ),
+            force=True,
+            animate=self.show_vertical_scrollbar,  # https://github.com/Textualize/textual/issues/2077
+        )
```

Comment:
Thanks for this! This gave me an idea for another issue I'm tackling! 😆

Reply:
Amusingly this reminded me that I needed to just turn it off (at Will's request! WILL WANTED NO ANIMATION!)